compiler/rustc_middle/src/ty/query/on_disk_cache.rs

```diff
@@ -851,14 +854,25 @@ where
             return TAG_INVALID_SPAN.encode(s);
         }
 
+        let (_, line_hi, col_hi) = match s.source_map.byte_pos_to_line_and_col(span_data.hi) {
+            Some(pos) => pos,
+            None => return TAG_INVALID_SPAN.encode(s),
+        };
+
         let len = span_data.hi - span_data.lo;
 
         let source_file_index = s.source_file_index(file_lo);
 
+        // We include both the end location and length, so that we can
+        // verify that the length we compute from the source map during
+        // deserialization is consistent with the serialized span.
+
         TAG_VALID_SPAN.encode(s)?;
         source_file_index.encode(s)?;
         line_lo.encode(s)?;
         col_lo.encode(s)?;
+        line_hi.encode(s)?;
+        col_hi.encode(s)?;
```

Comment:
If this is purely for assertion purposes, is it worth doing? Do we know if this change has any performance impact on its own?

Reply:
I can't say whether or not it's worth it, but it does have a performance impact. It accounts for most of the `tuple-stress` and `ucd` benchmark regressions. Here is the effect of removing the changes to `on_disk_cache.rs` (all of which are only necessary for the assert):

![image](https://user-images.githubusercontent.com/3668166/97356322-54dd2d00-1855-11eb-8c08-268dc9af57ad.png)

Reply:
Given the not insignificant impact of a single assert on at least one real-world crate (ucd), I'll go ahead and remove it and the extra encoding/decoding.

Reply:
Yeah, I think that's the right call.